cumulus/client/cirrus-executor/src/aux_schema.rs

```diff
@@ -0,0 +1,41 @@
+//! Schema for executor in the aux-db.
+
+use codec::{Decode, Encode};
+use sc_client_api::backend::AuxStore;
+use sp_blockchain::{Error as ClientError, Result as ClientResult};
+use sp_executor::ExecutionReceipt;
+use sp_runtime::traits::Block as BlockT;
+
+const EXECUTION_RECEIPT_KEY: &[u8] = b"execution_receipt";
+
+fn load_decode<Backend: AuxStore, T: Decode>(
+	backend: &Backend,
+	key: &[u8],
+) -> ClientResult<Option<T>> {
+	match backend.get_aux(key)? {
+		None => Ok(None),
+		Some(t) => T::decode(&mut &t[..])
+			.map_err(|e| {
+				ClientError::Backend(format!("Executor DB is corrupted. Decode error: {}", e))
+			})
+			.map(Some),
+	}
+}
+
+/// Write the execution receipt of a block to aux storage.
+pub(super) fn write_execution_receipt<Backend: AuxStore, Block: BlockT>(
+	backend: &Backend,
+	block_hash: Block::Hash,
+	execution_receipt: &ExecutionReceipt<Block::Hash>,
+) -> Result<(), sp_blockchain::Error> {
+	let key = (EXECUTION_RECEIPT_KEY, block_hash).encode();
+	backend.insert_aux(&[(key.as_slice(), execution_receipt.encode().as_slice())], [])
+}
+
+/// Load the execution receipt associated with a block.
+pub(super) fn load_execution_receipt<Backend: AuxStore, Block: BlockT>(
+	backend: &Backend,
+	block_hash: Block::Hash,
+) -> ClientResult<Option<ExecutionReceipt<Block::Hash>>> {
+	load_decode(backend, (EXECUTION_RECEIPT_KEY, block_hash).encode().as_slice())
+}
```

Comment:
Minor thing, but may be this can be generalized in some manner like this:
```rust
pub(super) struct FromAuxStore<Prefix, T>(PhantomData<Prefix, T>);

impl<Prefix, T> FromAuxStore<Prefix, T>
where
    Prefix: Get<Vec<u8>>, // `impl AsRef<u8>` should also do
    T: Decode,
{
    pub fn write<Backend: AuxStore, Block: BlockT>(
        backend: &Backend,
        block_hash: Block::Hash,
        value: &T,
    ) -> Result<(), sp_blockchain::Error> {
        let key = (Prefix::get().as_ref(), block_hash).encode();
        backend.insert_aux(&[(key.as_slice(), execution_receipt.encode().as_slice())], [])
    }
    // ...
}
```

Reply:
Getting some key can be  made async with some custom made future!

Reply:
Something like this:
```rust
// `Get<impl AsRef<u8>>` would suit better
impl<Prefix: Get<Vec<u8>>, T: Decode> FromAuxStore<Prefix, T> {
	pub fn load_async<Backend: AuxStore, Block: BlockT>(
		backend: &'_, Backend,
		block_hash: Block::Hash,
	) -> AuxLoadFuture<'_, Prefix, T, Backend, Block> {
		AuxLoadFuture { backend, block_hash }
	}
}

pub(crate) struct AuxLoadFuture<'a, Prefix, T, Backend, Block: BlockT> {
	backend: &'a Backend,
	block_hash: Block::Hash,
}

impl<'a, Prefix: Get<Vec<u8>>, T: Decode, Backend: AuxStore, Block: BlockT> Future for AuxLoadFuture {
	type Output = ClientResult<T>;
	fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
		match load_decode(self.backend, (Prefix::get().as_ref(), block_hash).encode_as_slice()) {
			Some(result) => Poll::Ready(result),
			None => Poll::Pending,
		}
	}
}
```

Reply:
Concerning the mere plain get/set function, I'm reluctant to introduce more abstractions before it's really necessary.

Reply:
Making it async is better than `sleep()`, is it ok for now @nazar-pc?

Reply:
>     2. I'm really not used to writing manual futures unless absolutely needed, but above future example isn't a future, it 
doesn't ever notify executor that it needs to be polled again

True, makes sense. Didn't think that you must to wake future. Then we can leave it as it is :)